pkg/labels/matcher.go

```diff
@@ -28,17 +28,18 @@ const (
 	MatchNotRegexp
 )
 
+var matchTypeToStr = [...]string{
+	MatchEqual:     "=",
+	MatchNotEqual:  "!=",
+	MatchRegexp:    "=~",
+	MatchNotRegexp: "!~",
+}
+
 func (m MatchType) String() string {
-	typeToStr := map[MatchType]string{
-		MatchEqual:     "=",
-		MatchNotEqual:  "!=",
-		MatchRegexp:    "=~",
-		MatchNotRegexp: "!~",
-	}
-	if str, ok := typeToStr[m]; ok {
-		return str
+	if m < 0 || int(m) > len(matchTypeToStr) {
```

Comment:
```suggestion
	if m < MatchEqual || m > MatchNotRegexp {
```

Reply:
Hm, not sure about this one: this change forces us to keep this method in sync with new MatchType values, while with current implementation we can add a new type and add the new string value for it without having to touch the `String()` method implementation. Plus, it relies on `MatchNotRegexp` being the last value in the enum, which is an extra cognitive load here IMO.

Reply:
Meh, not a strong opinion, plus my code was wrong since it had to be `int(m) >= len(matchTypeStr)` to be correct, so just applying it.

Reply:
The tests should ideally catch addition or deletion of matcher types. I see that there are no tests checking this string, but that is not for this PR to fix, hence merging :)